hudi-common/src/main/java/org/apache/hudi/common/schema/HoodieSchemaUtils.java

```diff
@@ -416,20 +416,115 @@ private static Option<Pair<String, HoodieSchemaField>> getNestedFieldInternal(Ho
           .map(field -> Pair.of(prefix + fieldName, field));
     } else {
       // Recursive case: nested field
-      if (nonNullableSchema.getType() != HoodieSchemaType.RECORD) {
-        return Option.empty();
-      }
-
       int dotIndex = fieldName.indexOf(".");
       String rootFieldName = fieldName.substring(0, dotIndex);
       String remainingPath = fieldName.substring(dotIndex + 1);
 
-      return nonNullableSchema.getField(rootFieldName)
-          .flatMap(rootField -> getNestedFieldInternal(
-              rootField.schema(),
-              remainingPath,
-              prefix + rootFieldName + "."
-          ));
+      // Handle RECORD types - standard field navigation
+      if (nonNullableSchema.getType() == HoodieSchemaType.RECORD) {
+        return nonNullableSchema.getField(rootFieldName)
+            .flatMap(rootField -> getNestedFieldInternal(
+                rootField.schema(),
+                remainingPath,
+                prefix + rootFieldName + "."
+            ));
+      }
+
+      // Handle ARRAY types - expect ".list.element" pattern
+      if (nonNullableSchema.getType() == HoodieSchemaType.ARRAY && "list".equals(rootFieldName)) {
+        if (!remainingPath.startsWith("element")) {
+          return Option.empty();  // Invalid path for ARRAY
+        }
+
+        // Skip "element" and get remaining path
+        String pathAfterElement = remainingPath.substring("element".length());
+        if (!pathAfterElement.isEmpty() && !pathAfterElement.startsWith(".")) {
+          return Option.empty();  // Invalid format
+        }
+        if (pathAfterElement.startsWith(".")) {
+          pathAfterElement = pathAfterElement.substring(1);
+        }
+
+        HoodieSchema elementSchema = nonNullableSchema.getElementType();
+        if (pathAfterElement.isEmpty()) {
+          // We've reached the end - return synthetic field for element
+          HoodieSchemaField syntheticField = HoodieSchemaField.of(
+              "element",
+              elementSchema,
+              null,  // doc
+              null   // defaultVal
+          );
+          return Option.of(Pair.of(prefix + rootFieldName + ".element", syntheticField));
+        } else {
+          // Continue navigating into element schema
+          return getNestedFieldInternal(
+              elementSchema,
+              pathAfterElement,
+              prefix + rootFieldName + ".element."
+          );
+        }
+      }
+
+      // Handle MAP types - expect ".key_value.key" or ".key_value.value" pattern
```

Comment:
Are these the only pattern for nested representation of map and array column? Parquet uses the same values?